rules/linux/credential_access_ssh_backdoor_log.toml

```diff
@@ -0,0 +1,61 @@
+[metadata]
+creation_date = "2020/12/21"
+maturity = "production"
+updated_date = "2020/12/21"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule identifies an SSH client or server process creating, or writing to, a known SSH backdoor log file.
+"""
+false_positives = ["Updates to approved and trusted SSH executables can trigger this rule."]
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Potential OpenSSH Backdoor Logging Activity"
+note = "Adversaries may modify SSH related binaries for persistence or credential access via patching sensitive functions to enable unauthorized access or to log SSH credentials for exfiltration."
+references = ["https://github.com/eset/malware-ioc/tree/master/sshdoor"]
+risk_score = 73
+rule_id = "f28e2be4-6eca-4349-bdd9-381573730c22"
+severity = "high"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Credential Access"]
+type = "eql"
+
+query = '''
+file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
+ (
+   wildcard(file.name, ".*", "~*") or
+   file.extension in ("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
+   wildcard(file.path, "/private/etc/*--", "/usr/share/*", "/usr/include/*", "/usr/local/include/*", "/private/tmp/*", "/private/var/tmp/*",
+                       "/usr/tmp/*", "/usr/share/man/*", "/usr/local/share/*", "/usr/lib/*.so.*", "/private/etc/ssh/.sshd_auth",
+                       "/usr/bin/ssd", "/private/var/opt/power", "/private/etc/ssh/ssh_known_hosts", "/private/var/html/lol", "/private/var/log/utmp", "/private/var/lib"
+	       )
+  )
```

Comment:
```suggestion
file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
 (
   file.name:(".*", "~*") or
   file.extension:("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
   file.path:(
     "/private/etc/*--", 
     "/usr/share/*", 
     "/usr/include/*", 
     "/usr/local/include/*", 
     "/private/tmp/*", 
     "/private/var/tmp/*",
     "/usr/tmp/*", 
     "/usr/share/man/*", 
     "/usr/local/share/*", 
     "/usr/lib/*.so.*", 
     "/private/etc/ssh/.sshd_auth",
     "/usr/bin/ssd", 
     "/private/var/opt/power", 
     "/private/etc/ssh/ssh_known_hosts", 
     "/private/var/html/lol", 
     "/private/var/log/utmp", 
     "/private/var/lib"
     )
  )
```

Reply:
removed the case sensitive `wildcard` function and made it a bit more readable. should look like below - 
```
file where event.type == "change" and process.executable in ("/usr/sbin/sshd", "/usr/bin/ssh") and
 (
   file.name:(".*", "~*") or
   file.extension:("in", "out", "ini", "h", "gz", "so", "sock", "sync", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9") or
   file.path:(
     "/private/etc/*--", 
     "/usr/share/*", 
     "/usr/include/*", 
     "/usr/local/include/*", 
     "/private/tmp/*", 
     "/private/var/tmp/*",
     "/usr/tmp/*", 
     "/usr/share/man/*", 
     "/usr/local/share/*", 
     "/usr/lib/*.so.*", 
     "/private/etc/ssh/.sshd_auth",
     "/usr/bin/ssd", 
     "/private/var/opt/power", 
     "/private/etc/ssh/ssh_known_hosts", 
     "/private/var/html/lol", 
     "/private/var/log/utmp", 
     "/private/var/lib"
     )
  )
```